### ISSUE TITLE
[Bug]: VoxCPM2 silently drops language / instructions / seed and hardcodes cfg_value, no normalize hook

### ISSUE THREAD

**Issue by simonxluo** [vllm-project/vllm-omni#4032]
### Your current environment

<details>
<summary>Environment</summary>

```text
- OS: TencentOS Server 4.4 (Linux x86_64)
- Hardware: NVIDIA GPU (any modern Ada/Hopper)
- Endpoint: POST /v1/audio/speech
```

`collect_env.py` is not relevant here — this is a logic/feature bug visible by code inspection on the current `main` branch, independent of the runtime.

</details>

### Your code version

<details>
<summary>The commit id or version of vllm</summary>

```text
0.20.x (any version compatible with the vllm-omni commit below)
```
</details>
<details>
<summary>The commit id or version of vllm-omni</summary>

```text
d4c139508fc490f856d8c22367cb85902caffb90 (main, as of 2026-06-01)
```
</details>

## Summary

When serving `openbmb/VoxCPM2` with `vllm serve --omni`, several user-facing parameters that the upstream VoxCPM model supports natively are silently dropped or hardcoded by the vllm-omni implementation. This causes degraded quality on Chinese-English mixed inputs (audible noise / "buzz" artefacts) and prevents users from controlling the voice style or language hint via the OpenAI-compatible API.

This is **not the same** as #3649 (which is about input validation gaps); the bug here is that valid parameters accepted by the request schema never reach the VoxCPM2 model.

### 1. `language` and `instructions` are accepted by the schema but never reach VoxCPM2

`OpenAICreateSpeechRequest` (in [`vllm_omni/entrypoints/openai/protocol/audio.py`](https://github.com/vllm-project/vllm-omni/blob/d4c13950/vllm_omni/entrypoints/openai/protocol/audio.py)) declares `language` and `instructions` as optional fields. They are stored on the request object but the VoxCPM2 branch of `_prepare_speech_generation` ([`serving_speech.py` L2615](https://github.com/vllm-project/vllm-omni/blob/d4c13950/vllm_omni/entrypoints/openai/serving_speech.py#L2615)) only forwards `text` / `ref_audio` / `ref_text` / `voice_profile` to `build_voxcpm2_prompt`.

Compare with the Qwen3-TTS branch (`_build_tts_params` ~L2073, L2132) which maps `request.language` → `params["language"]` and `request.instructions` → `params["instruct"]`.

Per the upstream gradio demo ([`examples/online_serving/text_to_speech/voxcpm2/gradio_demo.py` L508-510](https://github.com/vllm-project/vllm-omni/blob/d4c13950/examples/online_serving/text_to_speech/voxcpm2/gradio_demo.py#L508-L510)), VoxCPM2's control instruction is supposed to be prepended to `input` as `(instruction)text`. This convention is not applied server-side.

### 2. Generation hyper-parameters are hardcoded

In [`vllm_omni/model_executor/models/voxcpm2/voxcpm2_talker.py`](https://github.com/vllm-project/vllm-omni/blob/d4c13950/vllm_omni/model_executor/models/voxcpm2/voxcpm2_talker.py):

- `_inference_timesteps = 10` (L461)
- `_cfg_value = 2.0` (L462)
- `load_denoiser=False` (L444)

The official VoxCPM 2 docs (https://voxcpm.readthedocs.io/zh-cn/latest/usage_guide.html) recommend `cfg_value` between 1.0 and 3.0 with **1.5–1.6 specifically helping "buzzy / muddy" output** in long or mixed-language text. Users currently cannot tune this via the API.

### 3. No `normalize` (text normalization) hook

The official model exposes `normalize=True` for digit/date expansion. The OpenAI serving layer in vllm-omni has no equivalent — input text is fed verbatim to `tokenizer.encode`. This produces audible degradation on numeric and mixed-language input.

### 4. `seed` does not influence CFM noise

`request.seed` is forwarded to `sampling_params_list[0].seed`. However, VoxCPM2's `compute_logits` returns a deterministic 2-class logit (continue / stop), so seed has no effect at the sampler level. The actual non-determinism is in `_optimized_solve_euler` ([`voxcpm2_talker.py` L338](https://github.com/vllm-project/vllm-omni/blob/d4c13950/vllm_omni/model_executor/models/voxcpm2/voxcpm2_talker.py#L338)): `buffers.noise[:b].normal_()` draws from PyTorch global RNG without honouring `request.seed`. Reproducible generation is therefore impossible today.

### 5. Multi-character CJK tokens are split into single chars

`build_cjk_split_map` ([`voxcpm2_talker.py` L71-93](https://github.com/vllm-project/vllm-omni/blob/d4c13950/vllm_omni/model_executor/models/voxcpm2/voxcpm2_talker.py#L71-L93)) and the fail-fast assertion at L1199-1206 require every multichar CJK token to be split before being sent to prefill. This may diverge from the tokenizer behaviour the model was trained with and could be one root cause of the noise on mixed Chinese-English inputs.

## Repro

```bash
vllm serve openbmb/VoxCPM2 --omni --port 8000

curl -s -o noisy.pcm http://localhost:8000/v1/audio/speech \
  -H 'Content-Type: application/json' \
  -d '{
    "model": "openbmb/VoxCPM2",
    "input": "你好，我使用 Python 编程已经 5 年了。",
    "voice": "default",
    "language": "Chinese",
    "instructions": "A warm young woman, gentle voice",
    "seed": 42,
    "response_format": "pcm",
    "stream": true
  }'
```

**Expected**: stable, clean speech reflecting the requested language/style.
**Observed**: noticeable buzz/clicks on the mixed-language portion; identical output across different `seed` values; `language` / `instructions` have no audible effect.

## Suggested fix (sketch)

1. In the VoxCPM2 branch of `_prepare_speech_generation`, prepend `instructions` to the input text as `f"({instructions}){input}"` before tokenization, mirroring the gradio demo.
2. Add `cfg_value`, `inference_timesteps`, `normalize` to `OpenAICreateSpeechRequest` (or under `extra_params`). Plumb them through to `_run_cfm` / a tokenizer-side normalizer.
3. In `_optimized_solve_euler`, take a `torch.Generator` seeded from `request.seed` and pass it to `noise.normal_(generator=g)` so `seed` becomes meaningful.
4. Consider dropping the eager CJK split for VoxCPM2 (or making it opt-out) and verify against the model's training-time tokenizer behaviour.

I'd be happy to send a PR for items (1)–(3) if the maintainers agree on the direction. Item (4) needs guidance on whether the split was intentional for some reason I'm missing.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://vllm-omni.readthedocs.io), which can answer lots of frequently asked questions.
